packages/admin/cms-admin/src/generator/future/generateForm/generateFormField.ts

```diff
@@ -72,13 +84,58 @@ export function generateFormField({
 
     //TODO verify introspectionField.type is compatbile with config.type
 
-    const endAdornmentWithLockIconProp = `endAdornment={<InputAdornment position="end"><Lock /></InputAdornment>}`;
-    const readOnlyProps = `readOnly disabled`;
-    const readOnlyPropsWithLock = `${readOnlyProps} ${endAdornmentWithLockIconProp}`;
+    const gqlArgConfig =
+        !config.readOnly && createMutationType
+            ? (() => {
+                  const inputArg = createMutationType.args.find((arg) => arg.name === "input");
+                  if (!inputArg) throw new Error(`Field ${String(config.name)}: No input arg found`);
+                  let inputArgTypeRef = inputArg.type;
+                  if (inputArgTypeRef.kind === "NON_NULL") inputArgTypeRef = inputArgTypeRef.ofType;
+                  if (inputArgTypeRef.kind !== "INPUT_OBJECT") throw new Error(`Field ${String(config.name)}: input-arg is usually input-object.`);
+                  const inputArgTypeName = inputArgTypeRef.name;
+                  const inputArgType = gqlIntrospection.__schema.types.find((type) => type.name === inputArgTypeName);
+                  if (!inputArgType) throw new Error(`Field ${String(config.name)}: Input-Type ${inputArgTypeName} not found.`);
+                  if (inputArgType.kind !== "INPUT_OBJECT") {
+                      throw new Error(`Field ${String(config.name)}: Input-Type ${inputArgTypeName} is no input-object.`);
+                  }
+                  const inputArgField = inputArgType.inputFields.find((field) => field.name === name);
+
+                  let gqlArgField = inputArgField;
+                  let isInputArgSubfield = true;
```

Comment:
Wouldn't it be better if we'd use the inverse `isDedicatedResolver` arg here? Then we'd consistently use the dedicated resolver arg name.

Reply:
I think the "problem" is in generateForm.ts, where the mutation is generated. Then `isDedicatedResolver=false` would move the values into input-var. This is not what you would expect...

```
    if (addMode && createMutationType) {
        gqlDocuments[`create${gqlType}Mutation`] = `
            mutation Create${gqlType}(${
            gqlArgs.filter((gqlArg) => !gqlArg.isInputArgSubfield).length
                ? `${gqlArgs
                      .filter((gqlArg) => !gqlArg.isInputArgSubfield)
                      .map((gqlArg) => {
                          return `$${gqlArg.name}: ${gqlArg.type}!`;
                      })
                      .join(", ")}, `
                : ``
        }$input: ${gqlType}Input!) {
                ${createMutationType.name}(${
            gqlArgs.filter((gqlArg) => !gqlArg.isInputArgSubfield).length
                ? `${gqlArgs
                      .filter((gqlArg) => !gqlArg.isInputArgSubfield)
                      .map((gqlArg) => {
                          return `${gqlArg.name}: $${gqlArg.name}`;
                      })
                      .join(", ")}, `
                : ``
        }input: $input) {
                    id
                    updatedAt
                    ...${formFragmentName}
                }
            }
            \${${`${instanceGqlType}FormFragment`}}
        `;
    }
```